platform.js

```diff
@@ -147,14 +153,31 @@ class App {
   }
 
   asarApp () {
-    const asarOptions = common.createAsarOpts(this.opts)
-    if (!asarOptions) {
+    if (!this.asarOptions) {
       return Promise.resolve()
     }
 
     const dest = path.join(this.originalResourcesDir, 'app.asar')
-    debug(`Running asar with the options ${JSON.stringify(asarOptions)}`)
-    return pify(asar.createPackageWithOptions)(this.originalResourcesAppDir, dest, asarOptions)
+    debug(`Running asar with the options ${JSON.stringify(this.asarOptions)}`)
+
+    if (this.asarOptions.filename) {
+      const src = path.resolve(this.asarOptions.filename)
+      if (Object.keys(this.asarOptions).length > 1) {
+        common.warning('asar.filename has been specified, all other asar options will be ignored')
+      }
```

Comment:
This should have a test.

Reply:
what do you mean? Test that it warns? or test that the other options are ignored? If the later, how should that be done?

Reply:
Test that it warns. This is an example of doing that (although it should probably be genericized into a function):

https://github.com/electron-userland/electron-packager/blob/a7437b908e68e97b288173759d0538efe7798c82/test/mas.js#L18-L33

Reply:
done.